Makefile

```diff
@@ -35,10 +35,10 @@ build-all:
 	GOOS=darwin GOARCH=amd64 $(GO) build -o ${ARTIFACT_ROOT}/$(BINARY)-mac $(GOFLAGS)
 	GOOS=windows GOARCH=amd64 $(GO) build -o ${ARTIFACT_ROOT}/$(BINARY)-win.exe $(GOFLAGS)
 
-	tar -czvf ${ARTIFACT_ROOT}/$(BINARY)-linux-amd64-$(VERSION).tar.gz -C ${ARTIFACT_ROOT} $(BINARY)-linux-amd64
-	tar -czvf ${ARTIFACT_ROOT}/$(BINARY)-linux-arm64-$(VERSION).tar.gz -C ${ARTIFACT_ROOT} $(BINARY)-linux-arm64
-	tar -czvf ${ARTIFACT_ROOT}/$(BINARY)-mac-$(VERSION).tar.gz -C ${ARTIFACT_ROOT} $(BINARY)-mac
-	zip -j ${ARTIFACT_ROOT}/$(BINARY)-win-$(VERSION).zip ${ARTIFACT_ROOT}/$(BINARY)-win.exe
+	tar -czvf ${ARTIFACT_ROOT}/$(BINARY)-linux-amd64-$(VERSION).tar.gz Attribution.txt LICENSE -C ${ARTIFACT_ROOT} $(BINARY)-linux-amd64
```

Comment:
may need to copy license and attributions to ARTIFACT_ROOT .. the Jenkins pipeline fails to find the files to tar up. Or perhaps need to explicitly reference the git clone directory in the path?

Reply:
Today I was unable to explain myself why it fails 
The current directory within the jenkins server is the main edgex-cli directory and it contains all the files including Attribution.txt and LISENCE 
```
CURR_DIR=/w/workspace/edgex-cli/7
[2020-10-22T13:30:09.302Z] LS=Attribution.txt Dockerfile.build Jenkinsfile LICENSE Makefile README.md VERSION bin cmd config coverage.out go.mod go.sum main.go pkg res run_cmds.sh samples
```
(this is log is from some of the previous failed jenkins)

So it is strange for me why the following command fails: 
`tar -czvf ${ARTIFACT_ROOT}/$(BINARY)-linux-amd64-$(VERSION).tar.gz Attribution.txt LICENSE -C ${ARTIFACT_ROOT} $(BINARY)-linux-amd64`

Documentation of `tar` command says:
>man tar 
```
       -C, --directory=DIR
              Change to DIR before performing any operations.  This option is order-sensitive, i.e. it affects all options that follow.
```
 -C should takes effect after Attribution.txtx and LISENCE are found. It works on my local environment .. 

With one of the previous failed builds I tried to provide the absolute path towards these files ...an the tar was successfully created.  But within the resulted tar these two files wasn't places on the same lever as the executable. The directory structure was something like this
```
/edgex-cli-linux-amd64
/w/workspace/edgex-cli/3/
                          Attribution.txt
                          LISENSE
```
I want all the 3 file to be on the same level

It looks I am missing something ...

Reply:
@difince 

## Summary of the problem
The Jenkins builds happen in the context of docker container so Jenkins uses the BusyBox version of tar since we are using `golang:1.15-alpine` [base image](https://github.com/edgexfoundry/edgex-cli/blob/master/Dockerfile.build#L16) for the build. There are sometimes some discrepancies between the implementations of GNU packages and BusyBox packages. I was able to get things working locally by installing the [GNU tar version](https://pkgs.alpinelinux.org/package/v3.12/main/x86_64/tar) into the image by changing this line:

https://github.com/edgexfoundry/edgex-cli/blob/master/Dockerfile.build#L19

Before:

```bash
ARG ALPINE_PKG_BASE="make git zip"
```

After:

```bash
ARG ALPINE_PKG_BASE="make git zip tar"
```
## Fix

Update line 19 of `Dockerfile.build` and your build should pass.

## Local Testing

If you want to test locally how Jenkins will run the build, you can run this command:

```bash
$ docker build -t edgex-cli-ci-builder -f Dockerfile.build . && docker run --rm -v $(pwd):/ws -w /ws edgex-cli-ci-builder sh -c 'make build-all'
```

Reply:
@ernestojeda Great thanks!